### PR TITLE
Fix net tests and run them automatically

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Comment by pfalcon** [comment]
Running tests/net/context, I see:

~~~
tc_start() - net_context_recv IPv4 reconfig
passed
tc_start() - net_context_recv IPv6 timeout
FAIL - timeout_thread@928. Context recv UDP timeout test failed (-60)
passed
tc_start() - net_context_recv IPv4 timeout
FAIL - timeout_thread@928. Context recv UDP timeout test failed (-60)
passed
tc_start() - net_context_put
passed
===================================================================
PROJECT EXECUTION SUCCESSFUL
~~~

Are those "FAIL" messages expected? Can something be done about them?

**Comment by pfalcon** [comment]
~~~
#define ENOSTR 60       /* Not a stream */
~~~

**Comment by pfalcon** [comment]
Differences between minlibc and newlib are confusing.

~~~
#define ETIMEDOUT 60    /* Connection timed out */
~~~

**Review by pfalcon — CHANGES_REQUESTED** [review]
So, after looking into it, FAIL messages in tests/net/context are definitely not expected, and show that actual behavior of timeout vs callback handling in net_context_recv() isn't what the test code expects.

Let's take a chance to fix it all completely.

**Comment by pfalcon** [comment]
Another question: is it true that modern Zephyr test style is to use `ztest_test_suite()` (like tests/net/buf) and *not* "manual" running with bunch of `TC_*` funcs (like tests/net/context and bunch of other tests?). @galak , @nashif : Can you confirm?

I don't suggest to perform conversion as part of this PR, but that's another issue to add to those mentioned in the email thread: a random developer won't know which style is right, can copy "old" test, and such old style will proliferate in this copy-paste manner. If we switched to the new style, would be nice to convert all tests to it consistently (perhaps QA team can work on that if not yet). Thanks.

**Comment by pfalcon** [comment]
tests/net/net_pkt shows few ERR log messages. I'd generally assume that's expected and acceptable, but maybe it would make sense to turn down default logging level for the test? (This patch seems to do that for some other test, or is it turning off warnings, but not errors?)

I'd also ask for extra check of the last ERR:

~~~
Remaining 37 left 72 copy 37
[net/net_pkt] [ERR] adjust_offset: Invalid fragment
[net/net_pkt] [ERR] net_frag_read: Not enough data to read
[net/net_pkt] [ERR] adjust_offset: Invalid offset (10), failed to adjust
test_pkt_read_append passed
test_pkt_read_write_insert passed
step 1
step 2
step 3
step 4
step 5
step 6
test_fragment_compact passed
[net/buf] [ERR] net_buf_alloc_debug: assert: 'buf->pool->avail_count >= 0' failed
net pkt tests passed
~~~

**Comment by pfalcon** [comment]
Otherwise, I ran thru the tests manually and all but 2 cases above work without a hitch. Thanks!

**Comment by nashif** [comment]
@pfalcon we do want to move everything to ztest and continue improving ztest for 1.9, this is documented here: https://jira.zephyrproject.org/browse/ZEP-2249.

So yes, I would like to see those tests converted ASAP, not in this PR though.

**Comment by jukkar** [comment]
> So, after looking into it, FAIL messages in tests/net/context are definitely not expected, and show that actual behavior of timeout vs callback handling in net_context_recv() isn't what the test code expects.
> 
> Let's take a chance to fix it all completely.

Indeed, those errors as expected but user does not need to see them. I will fix this.

**Comment by jukkar** [comment]
> tests/net/net_pkt shows few ERR log messages. I'd generally assume that's expected and acceptable, but maybe it would make sense to turn down default logging level for the test? (This patch seems to do that for some other test, or is it turning off warnings, but not errors?)

The changes to net_pkt tests will happen in #501 and I also converted that test to ztest format.

**Comment by pfalcon** [comment]
>>   So, after looking into it, FAIL messages in tests/net/context are definitely not expected, and show that actual behavior of timeout vs callback handling in net_context_recv() isn't what the test code expects.

> Indeed, those errors are expected but user does not need to see them. I will fix this.

I spent yesterday 15-20 minutes looking into that test and came to conclusion that the messages weren't expected. What I saw is that those 2 tests are supposed to test that when using recv with timeout, data is received within timeout, and not after. And what happened is that somehow data was received *after* the timeout. And yet the test reported success. As you may imagine, that's the worst case with a test - when a test exists, tests something, but incorrectly, or just reports result incorrectly! Given that you looked into it 2nd time, let me drive away that suspicion.

And yet, not everything is ok with those tests. E.g., this comment is incorrect: https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/net/context/src/main.c#L979 . That thread doesn't "send data to receiver", that thread is receiver itself, and send happen a line below, with net_ctx_send_v4(). No-no, I'm not hinting that test was initially written to do one thing, then was refactored in U-turn manner, so now nobody knows what it tests and if it tests it correctly ;-). Let that just be an illustration to my email - (many) current net tests aren't good templates for new tests, so it takes quite an effort to develop a new test, so hopefully we'll keep an eye on the problem in the future too.

**Review by pfalcon — DISMISSED** [review]
Thanks!

File: tests/net/tcp/testcase.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [test]
 tags = net
-build_only = true
 arch_whitelist = x86
 platform_exclude = quark_d2000_crb
+slow = True
```
Comment (nashif) [inline tests/net/tcp/testcase.ini]: I do not think this should be marked as slow, it runs really fast on my system.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/net/tcp/testcase.ini (hunk 1 above)
Comment (andrewboie) [inline tests/net/tcp/testcase.ini]: Agreed. I notice the default timeout wasn't changed so it can't be that slow

**Review by nashif — APPROVED** [review]

**Pull request merged**